### PR TITLE
fix: MailToLink to account for no emails

### LINKED ISSUE #226: EmailLink Component
Discovery done with @cintnguyen:

The EmailLink component is covering for the use case of if there isn’t an email address provided. It’s being tightly coupled with Paragon’s [MailToLink](https://paragon-openedx.netlify.app/components/mailtolink/), returning null when the address is not provided. Seen [here](https://github.com/openedx/frontend-app-learner-dashboard/blob/master/src/components/EmailLink.jsx). It's being used throughout the code base and seems redundant. 

We addressed this in Paragon’s WG meeting for further clarification on if we should build out MailToLink in Paragon to handle this use case. You can find those meeting notes [here](https://openedx.atlassian.net/wiki/spaces/BPL/pages/3891429403/2023-10-18+Meeting+notes).

The takeaways from the WG are that If you’re using `MailtoLink`, we expect that an email address will exist to send. Currently, none of the following props are required: `to`, `cc`, and `bcc`. We could possibly make one of these required.

### PR DESCRIPTION
**Pull request by cintnguyen**
Resolves #226

Removed EmailLink component and refactored code to cover for the use case of not having an email provided. This led to also creating additional tests to cover for these snapshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "lint-fix": "fedx-scripts eslint --fix --ext .jsx,.js src/",
     "semantic-release": "semantic-release",
     "start": "fedx-scripts webpack-dev-server --progress",
-    "test": "TZ=GMT fedx-scripts jest --coverage --passWithNoTests",
+    "test": "TZ=GMT fedx-scripts jest --updateSnapshot --coverage --passWithNoTests",
```
Comment (brian-smith-tcril) [inline package.json:16]:
```suggestion
    "test": "TZ=GMT fedx-scripts jest --coverage --passWithNoTests",
```

File: src/containers/CourseCard/components/CourseCardBanners/CertificateBanner.test.jsx [hunk 2]
```diff
@@ -21,10 +21,10 @@ jest.mock('components/Banner', () => 'Banner');
...
 describe('CertificateBanner', () => {
   const props = { cardId: 'cardId' };
-  reduxHooks.usePlatformSettingsData.mockReturnValue({
-    supportEmail: 'suport@email',
-    billingEmail: 'billing@email',
-  });
+  // reduxHooks.usePlatformSettingsData.mockReturnValue({
+  //   supportEmail: 'suport@email',
+  //   billingEmail: 'billing@email',
+  // });
```
Comment (brian-smith-tcril) [inline src/containers/CourseCard/components/CourseCardBanners/CertificateBanner.test.jsx:27]:
```suggestion
```

**Review by brian-smith-tcril — COMMENTED** [review]
Overall this is looking great! I left a couple quick cleanup suggestions, but nothing major.

File: package.json (hunk 1 above)
Comment (cintnguyen) [reply package.json]: All set, thanks for the catch

File: src/containers/CourseCard/components/CourseCardBanners/CertificateBanner.test.jsx (hunk 2 above)
Comment (cintnguyen) [reply src/containers/CourseCard/components/CourseCardBanners/CertificateBanner.test.jsx]: All set, thanks for the catch!

**Comment by arbrandes** [comment]
@openedx/2u-aperture, can we get eyes on this one?  Thanks!

**Comment by arbrandes** [comment]
Oh, and @cintnguyen, mind rebasing?  Thanks!

**Comment by cintnguyen** [comment]
@arbrandes rebased, ci passed!

**Review by justinhynes — APPROVED** [review]
Looks good to me! Thanks for your work on this.

**Pull request merged**